modules/gdscript/gdscript.cpp

```diff
@@ -1183,6 +1253,83 @@ String GDScript::_get_gdscript_reference_class_name(const GDScript *p_gdscript)
 	return class_name;
 }
 
+GDScript *GDScript::_get_gdscript_from_variant(const Variant &p_variant) {
+	Variant::Type type = p_variant.get_type();
+	if (type != Variant::Type::OBJECT)
+		return nullptr;
+
+	Object *obj = p_variant;
+	if (obj == nullptr)
+		return nullptr;
+
+	return Object::cast_to<GDScript>(obj);
+}
+
+void GDScript::_get_dependencies(RBSet<GDScript *> &p_dependencies, const GDScript *p_except) {
```

Comment:
How would this function, and the overall approach, interact with a mixture of, say, GDScrip and C# classes?

Reply:
Good question. Like, if `A.gd` refers to `B.cs` and `B.cs` refers to `C.gd`? Mmm.

Reply:
I don't think it would affect this. C# cannot "preload" GDScript, so no C# constants can hold GDScripts. Neither is it possible to preload C# scripts, as it results to this:

```
SCRIPT ERROR: Parse Error: Preload file "res://HelloWorld.cs" has no resource loaders (unrecognized file extension).
          at: GDScript::reload (res://GameMenu.gd:5)
```

In the development of this PR, I mostly found out that it's constants that cause problems with cyclic dependencies, not dynamic loading of scripts.

So, I don't think it's a problem.